source/server/http/admin.cc

```diff
@@ -529,17 +544,62 @@ Http::Code AdminImpl::handlerClusters(absl::string_view url, Http::HeaderMap& re
   return Http::Code::OK;
 }
 
-// TODO(jsedgwick) Use query params to list available dumps, selectively dump, etc
-Http::Code AdminImpl::handlerConfigDump(absl::string_view, Http::HeaderMap& response_headers,
+Http::Code AdminImpl::handlerConfigDump(absl::string_view url, Http::HeaderMap& response_headers,
                                         Buffer::Instance& response, AdminStream&) const {
+  Http::Utility::QueryParams query_params = Http::Utility::parseQueryString(url);
+  const auto resource = resourceParam(query_params);
+  const auto mask = maskParam(query_params);
+
   envoy::admin::v2alpha::ConfigDump dump;
+  bool resource_found = false;
   for (const auto& key_callback_pair : config_tracker_.getCallbacksMap()) {
```

Comment:
it might be useful to split the code into two functions, one for searching through the map for the desired resource and one for dumping the entire config. That way you avoid the conditional logic inside of the loop, which should make this a bit easier to follow

Reply:
Good idea, I've split it up in a way that is IMO more clear.